### DISCUSSION TITLE
Looking for Feedback

### DISCUSSION THREAD

**Discussion by BetaHuhn** [WebCrateApp/webcrate#29] [Announcements]
👋 Hey guys, [Maximilian](https://github.com/betahuhn) here (creator/maintainer of WebCrate),

I'm currently looking to get more feedback about WebCrate itself and how you use it, specifically issues you are facing/things you like to see improved as well as new features you would like to see included in WebCrate!

It's an open discussion, share anything you like by commenting on this discussion! 💖

Additionally, I would like to talk to some of you individually in a 1:1 call and share some ideas I have for WebCrate and hear any feedback/thoughts you might have! If this sounds interesting to you, [select a time on Calendly](https://calendly.com/betahuhn/webcrate-feedback) or shoot me an email at [hi@webcrate.app](mailto:hi@webcrate.app)!

**Comment by ghost** [comment]
Love the app! But as it is, I'm currently not using it that much. I would love to see the option to change how links are displayed in crates. Right now, they are displayed in columns. It's a small thing, but it puts me off a bit. I use the crates as sort of a reading list, and as such, I'd like to be able to view the links as a one-column list.

Some other things you might want to consider putting:

1. Ability to reorder crates
2. Ability to mass select links. Really helpful when you're deleting lots of stuff or moving several links to different crates. Right now, I have to click on each one of them if I want to delete it.

**Reply by BetaHuhn** [reply]
Hey, thanks for the feedback!

An option to change how links are displayed is on my list, including showing image previews ([#31](https://github.com/WebCrateApp/webcrate/discussions/31)). When I implement that I will also add the ability to customize the column count.

Reordering crates is also something I was thinking about, but it will take a while until I work on that.

Will add mass selecting links and deleting/moving them at once to the list, but it will probably take a while until I work on that as well. Will focus on some of the other things first.

Thanks again for the suggestions! 😄

**Comment by 4ov** [comment]
Great app, i think you may want to add web share support to manifest, take a look [here](https://web.dev/web-share-target/).
it'd be great way to add links on the go on mobile devices

**Reply by BetaHuhn** [reply]
Hey, thanks for your suggestion and sorry for the late response!

I have never heard of the Web Share API, it definitely looks interesting. I'll add it to the list of planned features! I can't say when I will work on it since I'm focusing on a few other things first though.

**Reply by BetaHuhn** [reply]
This is included in version [1.4.1](https://github.com/WebCrateApp/webcrate/releases/tag/v1.4.1) :tada:

Added with [#46](https://github.com/WebCrateApp/webcrate/pull/46) by @hkamran80

**Comment by colorful-tones** [comment]
I'm loving this app, and thanks for all your hard work!

I would love the ability to be able to add attachments in the newly added notes area.

For example, here is my public [Professional Development](https://webcrate.colorfultones.deta.app/crate/public/6jZiHI8cIBh8WcwL) crate. I often link to Twitter posts, and a recent one that I wanted to bookmark linked off to a PDF. I _could_ link directly to the PDF, but prefer to keep the context of the Tweet as the bookmark, and would rather then attach the PDF referenced from Tweet within the entry. That would be super awesome! 👍

**Reply by BetaHuhn** [reply]
Hey Damon, thanks for taking the time to comment and for the nice words!

When you talk about attachments, do you mean attaching another link to it or directly attaching the PDF as a file?

What you could do right now is copy the link to the PDF and add it to the notes of the Twitter link. We could add support for links in the notes so you'd be able to click it. Would that work?

**Reply by colorful-tones** [reply]
> What you could do right now is copy the link to the PDF and add it to the notes of the Twitter link. We could add support for links in the notes so you'd be able to click it. Would that work?

I think enabling the Link button in the ProseMirror rich-text editor would be a great addition.

However, I also think that would be a worthwhile addition to be able to attach files to bookmark entries as well, but I know that is very likely a larger feature request.

**Reply by BetaHuhn** [reply]
Okay great! Will add the link button in the next release.

I don't know about files. I'll have to think about it and may come back to it at a later point. Thanks for the suggestion though!

**Reply by BetaHuhn** [reply]
The link button was added with [v1.4.0](https://github.com/WebCrateApp/webcrate/releases/tag/v1.4.0).

It should automatically linkify any URLs you paste and allows you to make any text a link via the bubble menu. when selecting.

**Comment by colorful-tones** [comment]
Another feature that would be super useful would be tagging. This adds another level of much needed hierarchy to organizing with taxonomies. 

Add a Link would have an additional field for Tags, and would autocomplete if tag was already used in the user's webcrate. Support for multiple tags would be ideal, and then the addition of user interface throughout the webcrate app to be able to filter by tags would be the icing on the cake.

Again, thanks for all your work. 👏

**Reply by colorful-tones** [reply]
I actually just came across #24 and this is exactly the same request.

**Reply by BetaHuhn** [reply]
Was about to comment that :)

This is definitely something I want to add in the short term, maybe even in the next release.

PS: Would love to have a short call (~30 min) with you to learn about how you use WebCrate and what can be improved as well as share some ideas I have for new features. Let me know if you'd be interested in that!
